cpp/src/gandiva/precompiled/extended_math_ops.cc

```diff
@@ -236,6 +237,25 @@ gdv_int64 round_int64(gdv_int64 num) { return num; }
 ROUND_DECIMAL(float32)
 ROUND_DECIMAL(float64)
 
+// rounds the number to the nearest integer
+#define BROUND_DECIMAL(TYPE)                       \
+  FORCE_INLINE                                     \
+  gdv_##TYPE bround_##TYPE(gdv_##TYPE num) {       \
+    const gdv_float64 round_num = round(num);      \
+    const gdv_float64 dif_num = round_num - num;   \
+    if ((dif_num != 0.5f) && (dif_num != -0.5f)) { \
```

Comment:
Should be 0.5f or 0.5d since dif_num is a double?

Reply:
Yes, it should, it will not have a support for float too.